### PR TITLE
Auth verification

### PR DESCRIPTION
**Pull request by mrdeveloperdude**
Refactored auth into several files.
Wrapped OAuth2Session class to get better access to debugging session expiration error that we are experiencing in production.
Added a few tests, cleaned up some logging and comments.

### CODE REVIEW

File: app/latigo/auth/session.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        return super(LatigoAuthSession, self).merge_environment_settings(url, proxies, stream, verify, cert)
+
+    def get_adapter(self, url):
+        logging.info(f"@@@ Latigo Session: get_adapter(url={url}).")
+        try:
+            adapter = super(LatigoAuthSession, self).get_adapter(url=url)
+            return adapter
+        except Exception as e:
+            logger.error(f"Could not get adapter for url='{url}': {e}")
+        #            raise e
```
Comment (ryanjdillon) [inline app/latigo/auth/session.py]: leftover comment

File: app/latigo/auth/session.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    def request(self, method, url, data=None, headers=None, withhold_token=False, client_id=None, client_secret=None, **kwargs):
+        logging.info(f"@@@ Latigo Session: request(method={method}).")
+        try:
+            response = super(LatigoAuthSession, self).request(method=method, url=url, data=data, headers=headers, withhold_token=withhold_token, client_id=client_id, client_secret=client_secret, **kwargs)
+            return response
+        except NewConnectionError as nce:
+            logger.error(f"Could not connect (method={method}, url='{url}'): {nce}")
+        except Exception as e:
+            logger.error(f"Could not perform request(method={method}, url='{url}'): {e}")
+        #            raise e
```
Comment (ryanjdillon) [inline app/latigo/auth/session.py]: leftover comment

File: tests/integration/test_auth.py [hunk 3]
```diff
@@ -34,3 +36,11 @@ def test_auth_verifier():
...
         logger.warning(f"Message was: '{message}'")
     assert res
+
+
+def test_session_signatures():
+    assert inspect.signature(OAuth2Session) == inspect.signature(LatigoAuthSession)
+    assert inspect.signature(OAuth2Session.__init__) == inspect.signature(LatigoAuthSession.__init__)
+    oas2 = OAuth2Session()
+    las = LatigoAuthSession()
+    assert inspect.signature(oas2.request) == inspect.signature(las.request)
```
[the remark this answers is not in the document]
Comment (mrdeveloperdude) [reply tests/integration/test_auth.py]: Cool! I will def look at that!

[the remark this answers is not in the document]
Comment (mrdeveloperdude) [reply tests/integration/test_auth.py]: I will definitely incorporate this once the testing gets serious. Reverting for now.

File: app/latigo/auth/session.py (hunk 1 above)
Comment (mrdeveloperdude) [reply app/latigo/auth/session.py]: Fixed

File: app/latigo/auth/session.py (hunk 2 above)
Comment (mrdeveloperdude) [reply app/latigo/auth/session.py]: Fixed

**Comment by mrdeveloperdude** [comment]
OK

**Pull request merged**